app/src/main/java/com/mensinator/app/CalendarScreen.kt

```diff
@@ -501,172 +178,378 @@ fun CalendarScreen(modifier: Modifier) {
+        )
+        HorizontalDivider(
+            color = MaterialTheme.colorScheme.primary,
+            thickness = 2.dp,
+            modifier = Modifier.padding(horizontal = 16.dp)
+        )
     }
 }
 
-//Return true if selected dates
-fun containsPeriodDate(selectedDates: Set<LocalDate>, periodDates: Map<LocalDate, Int>): Boolean {
-    return selectedDates.any { selectedDate ->
-        periodDates.containsKey(selectedDate)
+/**
+ * Display a day in the calendar.
+ */
+@Composable
+fun Day(day: CalendarDay, selectedDates: MutableState<Set<LocalDate>>, actualPeriodDates: Map<LocalDate, Int>,
+        actualOvulationDates: Set<LocalDate>, actualSymptomDates: Set<LocalDate>, ovulationPredictionDate: LocalDate,
+        periodPredictionDate: LocalDate) {
+
+    val colorMap = ColorSource.getColorMap(isDarkMode())
+    val dbHelper: IPeriodDatabaseHelper = koinInject()
+
+    val showCycleNumbersSetting =
+        dbHelper.getSettingByKey("cycle_numbers_show")?.value?.toIntOrNull() ?: 1
+
+    if (day.position != DayPosition.MonthDate) {
+        // Exclude dates that are not part of the current month
+        Box(
+            modifier = Modifier
+                .aspectRatio(1f) // Maintain grid structure with empty space
+        )
+        return
+    }
+
+    //Colors
+    val periodColor =
+        dbHelper.getSettingByKey("period_color")?.value?.let { colorMap[it] } ?: colorMap["Red"]!!
+    val selectedColor = dbHelper.getSettingByKey("selection_color")?.value?.let { colorMap[it] }
+        ?: colorMap["LightGray"]!!
+    val nextPeriodColor =
+        dbHelper.getSettingByKey("expected_period_color")?.value?.let { colorMap[it] }
+            ?: colorMap["Yellow"]!!
+    val ovulationColor = dbHelper.getSettingByKey("ovulation_color")?.value?.let { colorMap[it] }
+        ?: colorMap["Blue"]!!
+    val nextOvulationColor =
+        dbHelper.getSettingByKey("expected_ovulation_color")?.value?.let { colorMap[it] }
+            ?: colorMap["Magenta"]!!
+
+    val backgroundColor = when {
+        day.date in selectedDates.value -> selectedColor
+        day.date in actualPeriodDates.keys -> periodColor
+        day.date.isEqual(periodPredictionDate) -> nextPeriodColor
+        day.date in actualOvulationDates -> ovulationColor
+        day.date.isEqual(ovulationPredictionDate) -> nextOvulationColor
+        else -> Color.Transparent
+    }
+
+
+    val borderSize = when {
+        day.date.isEqual(LocalDate.now()) -> 1.dp
+        else -> 0.dp
+    }
+
+    val borderColor = when {
+        day.date.isEqual(LocalDate.now()) -> Color.LightGray
+        else -> Color.Transparent
+    }
+
+    val fontStyleType = when {
+        day.date.isEqual(LocalDate.now()) -> FontWeight.Bold
+        else -> FontWeight.Normal
+
+    }
+
+    //Dates to track
+    val isSelected = day.date in selectedDates.value
+    val hasSymptomDate = day.date in actualSymptomDates
+
+    Box(
+            modifier = Modifier
```

Comment:
Ripple caused by onClick is not clipped to the background. Wrap this in Surface or add .clip() modifier